wurst/_handles/Destructable.wurst

```diff
@@ -69,6 +69,81 @@ public function destructable.setOccluderHeight(real height)
 public function destructable.getName() returns string
 	return GetDestructableName(this)
 
+// Extensions
+
+tuple gate(destructable killable)
+/** Used to open a destructable gate */
+public function gate.open()
+	if (GetDestructableLife(this.killable) > 0)
+		KillDestructable(this.killable)
+
+/** Used to close a destructable gate */
+public function gate.close()
+	this.open()
+	DestructableRestoreLife(this.killable, GetDestructableMaxLife(this.killable), true)
+	SetDestructableAnimation(this.killable, "stand")
+
+/** Used to destroy a destructable gate */
+public function gate.kill()
```

Comment:
Any reason this should be public?

Reply:
Yes, the gate has "death" and "death alternate" (aka open) animation.
Kill triggers the "destroy" animation, while the regular open triggers the alternative animation. Different use cases.